vip-api/src/main/java/fr/insalyon/creatis/vip/api/utils/KeycloakRefreshUtils.java

```diff
@@ -0,0 +1,70 @@
+package fr.insalyon.creatis.vip.api.utils;
+
+import fr.insalyon.creatis.vip.api.KeycloakProperties;
+import fr.insalyon.creatis.vip.api.exception.ApiException;
+import org.keycloak.representations.AccessTokenResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.*;
+import org.springframework.stereotype.Component;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author alaeessaki this class is only for testing the refresh mechanism
+ */
+@Component
+public class KeycloakRefreshUtils {
+
+    private final Environment env;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private static final byte ALLOWED_ATTEMPTS = 2;
+
+    @Autowired
+    KeycloakRefreshUtils(Environment env){
+        this.env = env;
+    }
+
+    public ResponseEntity<AccessTokenResponse> refreshToken(String offlineToken) throws ApiException {
+        int attempt = 1; // 2 attempts max
+        RestTemplate restTemplate = new RestTemplate();
+
+        /**
+         *
+         * Creating the http headers with the values needed for the refreshing action
+         */
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+
+        map.add("client_id", env.getRequiredProperty(KeycloakProperties.KEYCLOAK_CLIENT_ID));
```

Comment:
Thanks for adding these properetis.
Now you should use the `SpringConfigServer` class to get those properties as it is done this way in all the VIP code.

Reply:
Hello Axel, sorry for the late reply ! 
but to use `SpringConfigServer` it means I should add the properties in the `vip.conf` file ? while I used vip-api.conf like the carmin properties